packages/orchestrator/internal/server/main.go

```diff
@@ -59,22 +68,78 @@ func New() (*grpc.Server, error) {
 		return nil, fmt.Errorf("failed to create network pool: %w", err)
 	}
 
-	s := grpc.NewServer(
-		grpc.StatsHandler(e2bgrpc.NewStatsWrapper(otelgrpc.NewServerHandler())),
-		grpc.ChainUnaryInterceptor(
-			recovery.UnaryServerInterceptor(),
-		),
-	)
+	// BLOCK: initialize services
+	{
+		srv.dns = dns.New()
+		srv.grpc = grpc.NewServer(
+			grpc.StatsHandler(e2bgrpc.NewStatsWrapper(otelgrpc.NewServerHandler())),
+			grpc.ChainUnaryInterceptor(
+				recovery.UnaryServerInterceptor(),
+			),
+		)
+
+		srv.server = &server{
+			tracer:        otel.Tracer(ServiceName),
+			dns:           srv.dns,
+			sandboxes:     smap.New[*sandbox.Sandbox](),
+			networkPool:   networkPool,
+			templateCache: templateCache,
+		}
+	}
+
+	orchestrator.RegisterSandboxServiceServer(srv.grpc, srv.server)
+	grpc_health_v1.RegisterHealthServer(srv.grpc, health.NewServer())
+
+	return srv, nil
+}
+
+// Start launches
+func (srv *Service) Start(context.Context) error {
+	if srv.server == nil || srv.dns == nil || srv.grpc == nil {
+		return errors.New("orchestrator services are not initialized")
+	}
+
+	go func() {
+		log.Printf("Starting DNS server")
+		if err := srv.dns.Start("127.0.0.4", 53); err != nil {
+			log.Panic(fmt.Errorf("Failed running DNS server: %w", err))
+		}
+	}()
+
+	// the listener is closed by the shutdown operation
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", srv.port))
+	if err != nil {
+		return fmt.Errorf("failed to listen on port %d: %w", srv.port, err)
```

Comment:
if this happens, wouldn't this result in panic? `srv.closer.op` won't be defined, but `Close` will be called